pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -452,6 +452,22 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 			pipelineMeta.Namespace, pipelineMeta.Name, err)
 		return controller.NewPermanentError(err)
 	}
+	// If the pipeline spec was fetched from a referenced pipeline then we also need to validate
+	// the parameter and workspace usage because propagation is not allowed.
+	if pr.Spec.PipelineRef != nil {
```

Comment:
Actually, I think it's not redundant-- it appears this PR adds this validation for referenced pipelines, but it looks like we don't have this validation for inline pipelines based on usages of these functions. For that reason and also for consistency with the taskrun reconciler I think it would make sense to call this validation for all pipelinespecs post propagation.

Reply:
Sounds good! I will leave the call here as is so that we can validate that the referenced pipelines were explicit in declaring params and workspaces. I will also call this post propagation (before creation of the taskrun) regardless of whether the pipeline was referenced since we expect propagation to have completed by then.

Reply:
maybe just post propagation for now, and we can add the extra validation check if needed? Do you think the extra check will help?

Reply:
We do need one pre-propagation because we need to reject referenced pipelines that have `implicit spec`. Otherwise, we will end up supporting propagated params for referenced pipelines too which goes against Issue https://github.com/tektoncd/pipeline/issues/6670 (what we are trying to fix here).

Reply:
ah you're right, my bad!